test/module/ecdsa-registry-module.specs.ts

```diff
@@ -0,0 +1,156 @@
+                    return address(this);
+                }
+            }
+            `;
+            const randomContract = await deployContract(deployer,randomContractCode);
+
+
+            let EcdsaOwnershipSetupData = EcdsaRegistryModule.interface.encodeFunctionData(
+                "initForSmartAccount",
+                [await randomContract.address]
+              );
+
+            // Getting the address of the to be deployed SA does not get revert as it's just a bunch of abi.encode operations
+            const expectedSmartAccountAddress = await SAFactory.getAddressForCounterFactualAccount(EcdsaRegistryModule.address,EcdsaOwnershipSetupData,smartAccountDeploymentIndex);
+            // This should revert
+            await expect(SAFactory.deployCounterFactualAccount(EcdsaRegistryModule.address,EcdsaOwnershipSetupData,smartAccountDeploymentIndex)).to.reverted;
+            // Extra checks
+            // smartAccountOwners mapping should point to address(0)
+            expect( await EcdsaRegistryModule.smartAccountOwners(expectedSmartAccountAddress)).to.be.equal(AddressZero);
+        });
+
+        // Listen for AlreadyInitedForSmartAccount(msg.sender) error
+        it("Reverts when calling again after initialization", async()=>{
+            const {SAFactory,EcdsaRegistryModule,entryPoint} = await setupTests();
+
+            let EcdsaOwnershipSetupData = EcdsaRegistryModule.interface.encodeFunctionData(
+                "initForSmartAccount",
+                [await smartAccountOwner.getAddress()]
+              );
+
+            const expectedSmartAccountAddress = await SAFactory.getAddressForCounterFactualAccount(EcdsaRegistryModule.address,EcdsaOwnershipSetupData,smartAccountDeploymentIndex);
+            await SAFactory.deployCounterFactualAccount(EcdsaRegistryModule.address,EcdsaOwnershipSetupData,smartAccountDeploymentIndex);
+            const userSA = await hre.ethers.getContractAt("SmartAccount", expectedSmartAccountAddress);
+
+            await deployer.sendTransaction({
+                to: userSA.address,
+                value: ethers.utils.parseEther("10"),
+            });
+
+            // Send txs to ECDSARegistryModule.initForSmartAccount and check for revert
+            // Construct userOp
+            const txnData = EcdsaRegistryModule.interface.encodeFunctionData(
+                "initForSmartAccount",
+                [bob.getAddress()],
+            );
+            const userOp = await fillAndSign(
+                {
+                    sender: userSA.address,
+                    callData: txnData
+                },
+                smartAccountOwner,
+                entryPoint,
+                'nonce'
+            );
+            let signatureWithModuleAddress = ethers.utils.defaultAbiCoder.encode(
+                ["bytes", "address"],
+                [userOp.signature, EcdsaRegistryModule.address]
+            );
+            userOp.signature = signatureWithModuleAddress;
```

Comment:
1. You can use `makeEcdsaModuleUserOp` helper function rather than constructing the signature by yourself.
2. This code makes EntryPoint trying to call `userSA.initForSmartAccount()`. However, `initForSmartAccount` method is located in the module, not in the SA. 
So you need to construct the userOp.calldata for the `executeCall` method of SmartAccount that will call `ECDSAModule.initForSmartAccount()`